### PR TITLE
Add `ManaAllotment` in Node.js

### PR DESCRIPTION
**Pull request by abdulmth**
fixes https://github.com/iotaledger/iota-sdk/issues/914

I think that's all what we need in Node.js since we don't need the `ManaAllotments` as in Rust because it can be just represented as `ManaAllotment[]`.

### CODE REVIEW

File: bindings/nodejs/lib/types/block/mana-allotment.ts [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+ */
+class ManaAllotment {
+    /**
+     * The Account to allot Mana to.
+     */
+    readonly accountId: AccountId;
+    /**
+     * The Amount of Mana to allot.
+     */
+    readonly mana: bigint;
```
Comment (thibault-martinez) [inline bindings/nodejs/lib/types/block/mana-allotment.ts]: Same discussion as before but I actually still don't understand. Why not u64? What is different between this mana and the one on the Output that is a u64?

Comment (abdulmth) [reply bindings/nodejs/lib/types/block/mana-allotment.ts]: I thought we wanted to change all of them to `bigint` 😅

Comment (thibault-martinez) [reply bindings/nodejs/lib/types/block/mana-allotment.ts]: But we just introduced u64 a couple of days ago?

Comment (abdulmth) [reply bindings/nodejs/lib/types/block/mana-allotment.ts]: Changed it

**Review by thibault-martinez — APPROVED** [review]

**Review by qrayven — APPROVED** [review]

**Review by Thoralf-M — APPROVED** [review]

**Review by kwek20 — COMMENTED** [review]
We could add a type alias Mana Allotments  anyway to mimic rust code more.

**Comment by thibault-martinez** [comment]
> We could add a type alias Mana Allotments anyway to mimic rust code more.

Have we done that with the other list/set types? Like Inputs, Outputs, NativeTokens, ... ?

**Review by DaughterOfMars — APPROVED** [review]

**Comment by abdulmth** [comment]
> We could add a type alias Mana Allotments anyway to mimic rust code more.

In Rust it's necessary, but I'm not sure how the type alias would bring any benefit to TS. I would just use a normal array.

**Pull request merged**